v1/ao/internal/config/env.go

```diff
@@ -55,13 +55,16 @@ func loadEnvsInternal(c interface{}) {
 				continue
 			}
 
-			setField(c, "Set", field, stringToValue(envVal, fieldV.Type()))
+			val, err := stringToValue(envVal, fieldV.Type())
+			if err == nil {
```

Comment:
I got a Go noob question 😅 ! For the cases that there's an error, what would the `val` be? (`nil` or some default value of that type?)

Since we are no longer calling `setField` on error, would it change any behavior? For example, would some fields be set to some default value on error before but left blank after this change?

Reply:
Normally if err is not nil the val should be considered as undefined and should not be used. For this case it depends on what value the `strconv` functions return. 

> Since we are no longer calling setField on error, would it change any behavior?

Yes and it's what we want, as previously if there is an error the value (mostly the default value of that type) will still be assigned which is incorrect.